include/nbl/video/alloc/SubAllocatedDescriptorSet.h

```diff
@@ -75,51 +82,55 @@ class SubAllocatedDescriptorSet : public core::IReferenceCounted
 			auto& range = m_allocatableRanges[i];
 			if (range.reservedSize == 0)
 				continue;
-
 			auto ptr = reinterpret_cast<const uint8_t*>(core::address_allocator_traits<AddressAllocator>::getReservedSpacePtr(*range.addressAllocator));
+			range.addressAllocator->~PoolAddressAllocator();
```

Comment:
this is not how a shared or unique pointer works, you'll crash

just assign nullptr and it will kick in